chain/src/main/scala/org/bitcoins/chain/models/BlockHeaderDAO.scala

```diff
@@ -207,10 +207,14 @@ case class BlockHeaderDAO()(implicit
     table.filter(header => header.height >= from && header.height <= to).result
   }
 
-  def findAllBeforeTime(time: UInt32): Future[Vector[BlockHeaderDb]] = {
-    val query = table.filter(_.time < time)
+  def findBeforeTime(time: UInt32): Future[Option[BlockHeaderDb]] = {
```

Comment:
Probably should be rename to something like `findClosestBeforeTime()` or something like that? Otherwise I would think `findBeforeTime()` would return all headers before a specific time